torch/csrc/dynamo/python_compiled_autograd.cpp

```diff
@@ -449,20 +461,17 @@ variable_list compiled_autograd(
     call->node->release_variables();
   }
 
-  THPObjectPtr inputs(THPVariable_WrapList(compiler_call.inputs));
+  THPObjectPtr inputs(THPVariable_WrapList(compiler_call.tensor_args.inputs));
   THPObjectPtr sizes(wrap_int_list(compiler_call.dyn_size_inputs));
   THPObjectPtr hooks(convert_hook_list(compiler_call.hooks));
   THPObjectPtr pyresult(check(PyObject_CallFunctionObjArgs(
       cache->compiled_fn.get(), inputs.get(), sizes.get(), hooks.get(), NULL)));
   variable_list outputs = THPVariable_UnpackList(pyresult);
   if (accumulate_grad) {
-    TORCH_INTERNAL_ASSERT(
-        outputs.size() == compiler_call.set_grad_targets.size());
+    TORCH_INTERNAL_ASSERT(outputs.size() == cache->output_grad_targets.size());
     for (const auto i : c10::irange(outputs.size())) {
-      // TODO(jansel): does this one need to be an inplace copy?  if so it
-      // should go in the graph
-      at::Tensor& grad = compiler_call.set_grad_targets[i].mutable_grad();
-      grad = outputs[i];
+      compiler_call.tensor_args.inputs[cache->output_grad_targets[i]]
+          .mutable_grad() = outputs[i];
```

Comment:
In the inplace case, we have `compiler_call.tensor_args.inputs[cache->output_grad_targets[i]].mutable_grad().arg.unsafeGetTensorImpl() == outputs[i].arg.unsafeGetTensorImpl()` leading to the grad field being preserved.
Can we mention that in the comment above?

Reply:
In the inplace case the callback lambda is never called, and this code isn't run.  The graph just contains a `add_` op in it on one of its inputs (and returns nothing).

Notice how there is no call to `update_grad()` in this case:
https://github.com/pytorch/pytorch/blob/4f76be8c77acd933c580512cdb955a6cd6e95a4a/torch/csrc/autograd/functions/accumulate_grad.h#L209

Reply:
Ho right.
Could we have some comment that explains how accumulate grad is handled and the different cases of what might happen?

Reply:
Done